amethyst_assets/src/source/dir.rs

```diff
@@ -27,7 +27,7 @@ impl Directory {
 
     fn path(&self, s_path: &str) -> PathBuf {
         let mut path = self.loc.clone();
-        path.push(s_path);
+        path.push(Path::new(s_path).iter().collect::<PathBuf>());
```

Comment:
Doesn't `PathBuf::from(s_path)` work?

Reply:
Ah, you already answered that.

Reply:
Can we do the following?

```rust
path.extend(Path::new(s_path).iter());
```

That would save us some allocations.